src/main/java/teammates/ui/webapi/action/GetStudentProfileAction.java

```diff
@@ -72,6 +72,12 @@ public ActionResult execute() {
             studentProfile = StudentProfileAttributes.builder(studentId).build();
         }
 
+        // If student requesting and is not the student's own profile, hide some fields
+        if (userInfo.isStudent && studentId.equals(userInfo.id)) {
```

Comment:
I think you should change `StudentProfileData` instead. See how `FeedbackSessionData` is done.

Reply:
Thanks, I've added a hide feature to `StudentProfileData`

Reply:
Hey @xpdavid, is the fix for this alright? Thanks!